flow/Trace.h

```diff
@@ -333,6 +333,7 @@ struct TraceableStringImpl : std::true_type {
 				result.push_back(base16Char(byte));
 			}
 		}
+		ASSERT(result.size() == result.capacity());
```

Comment:
Not sure if this assert is worth having in the production code. Might consider making it a `TEST_ASSERT` (I think that's what that does anyway)

Reply:
Looks like there is no `TEST_ASSERT` in the codebase now and I'm pretty sure ASSERT would be ignored in production

Reply:
I believe `ASSERT_WE_THINK` is the relevant test macro to test something only in simulation.